packages/react-forms/src/utils/value-helpers.ts

```diff
@@ -124,3 +161,59 @@ export function ValueOfType<TRequiredType>(
     }
     return true;
 }
+
+export function RecordifyModifierValue<TValue extends ModifierValue>(value: TValue): ModifierValueRecord {
+    return recordify<ModifierValue, ModifierValueRecord>(value);
+}
+
+/**
+ * Temporary functions for ImmutableJS v4 substitutions
+ */
+
+const IS_ITERABLE_SENTINEL: string = "@@__IMMUTABLE_ITERABLE__@@";
+const IS_RECORD_SENTINEL: string = "@@__IMMUTABLE_RECORD__@@";
+function isImmutable(maybeImmutable: any): boolean {
+    return (isCollection(maybeImmutable) || isRecord(maybeImmutable)) &&
+        !maybeImmutable.__ownerID;
+}
+
+function isValueImmutable(value: any): boolean {
+    return value == null || typeof value !== "object" || isImmutable(value);
+}
+
+function throwIfMutable(value: any, errorMessage: string): void {
+    if (!isValueImmutable(value)) {
+        throw new Error(errorMessage);
+    }
+}
+
+function isCollection(maybeCollection: any): boolean {
+    return !!(maybeCollection && maybeCollection[IS_ITERABLE_SENTINEL]);
+}
+
+function isRecord(maybeRecord: any): boolean {
+    return !!(maybeRecord && maybeRecord[IS_RECORD_SENTINEL]);
+}
+
+function ensureModifierValueImmutability(value: ModifierValue): ModifierValue {
+    if (value == null || isImmutable(value)) {
+        return value;
+    }
+
+    return {
+        Value: ensureImmutability(value.Value),
+        TransitionalValue: ensureImmutability(value.TransitionalValue)
+    };
+}
+
+function ensureImmutability(value: any): any {
```

Comment:
Yeah. I couldn't find a common type, because it's not as easy as `Readonly<T>` :/